flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdWindowProperties.scala

```diff
@@ -252,13 +254,40 @@ class FlinkRelMdWindowProperties private extends MetadataHandler[FlinkMetadata.W
   }
 
   def getWindowProperties(
-      rel: StreamPhysicalWindowAggregate,
+      rel: StreamPhysicalWindowAggregateBase,
       mq: RelMetadataQuery): RelWindowProperties = {
+    rel match {
```

Comment:
How about just overriding three methods of the concrete subclasses here? so that we can avoid forced type conversions here